atlp-rwanda/e-commerce-lydia-32-bn
# Ft password expiration prompt#187570893

### What does this PR do?
This PR implements a feature that prompts users to change their passwords every X amount of time since their last manual update. It includes functionality to notify users when their password is about to expire and enforces the update before allowing further operations.
### Description of Task to be completed?

- Add a field in the user profile model to store the timestamp of the last password update.
- Implement a cron job to iterate through all users and check if their password expiration time has reached.
- Use Node events to handle the password update event.
- Send notifications to users to prompt them to update their passwords.
- Enforce password updates by preventing further operations if the password update prompt is ignored.
- Use environment variables to configure the time for password expiration.

### How should this be manually tested?

1. Set up the database and ensure the last_password_update field is added to the user profile model.
2. Configure the environment variable PASSWORD_EXPIRATION_DAYS.
3. Run the application and create/update user passwords.
4. Verify that notifications are sent to users when their passwords are about to expire.
5. Check that users are prevented from performing further operations if they ignore the password update prompt.
6. Ensure the timer resets upon successful password updates.

### Any background context you want to provide?
This feature is crucial for maintaining the security of the application by enforcing regular password updates, thereby reducing the risk of security breaches.

### What are the relevant pivotal tracker stories
[https://www.pivotaltracker.com/n/projects/2703469](url)
